### DISCUSSION TITLE
Vim multi-cursor support

### DISCUSSION THREAD

**Discussion by kierangilliam** [zed-industries/zed#6825] [Feature Requests]
I saw in [this demo](https://zed.dev/img/post/assistant/demo.webm) that Zed supports multiple cursors in insert mode. Is there compatibility for multiple cursors in command mode? In VSCode, I often use multi-cursors along with `f`, `ciw`, etc. 

Example:

[Kapture 2024-01-26 at 19.10.27.webm](https://github.com/zed-industries/zed/assets/18583882/ff943f2f-457e-4794-b0c8-8b14358bb719)

I can't get this to work in Zed out of the box. Is there an option to enable this?

**Comment by mrnugget** [comment]
> Is there compatibility for multiple cursors in command mode?

You mean `normal` mode, right? Because command mode is the one triggered with `:`.


And yes, you can use the multi-cursors in normal mode: 

https://github.com/zed-industries/zed/assets/1185253/eccd307d-62a7-4a52-a775-ef501ee82464


There are still a lot of things to solve with regards to how visual mode and multi-cursor should interact, but yes, it works :)

**Reply by kierangilliam** [reply]
Correct! I do mean normal mode. Thanks for clarifying that. CMD D is great. 

Is there a way to create multiple cursors for lines that do not share an identifier? For example:

https://github.com/zed-industries/zed/assets/18583882/927e16e4-7b5d-4bcb-8591-1aa15f70428a

**Reply by SomeoneToIgnore** [reply]
There's an "editor: split selection into lines" action that should help.

**Reply by kenandotfyi** [reply]
Not exactly what you show here, but you could also use the visual block mode with Ctrl + V and then go into multi-cursor mode.

**Reply by seb-hyland** [reply]
Sorry I know this is kind of an old discussion, but is there any way to go from visual block into multi-cursor while dropping the selection?

**Reply by kenandotfyi** [reply]
Sorry I haven't seen this before...

What do you mean by dropping the selection?

**Comment by ohroy** [comment]
why
```
Select line | Editor | ⌘ + L
```
not work with multi cursor?

**Reply by marcospb19** [reply]
You should create a dedicated issue if you think something is broken and you want that fixed :)

**Comment by socketbox** [comment]
Is multi-cursor supported in Visual Block mode (ctrl-v)? When I attempt to insert a space in front of a selected column, nothing happens.

**Reply by kenandotfyi** [reply]
I do Shift + A after selecting a column to append text. Shift + I for insert...

**Reply by gpxl-dev** [reply]
I added this keybind to get the same with just i and a 
```json
  {
    "context": "Editor && vim_mode == visual",
    "bindings": {
      "i": "vim::InsertBefore",
      "a": "vim::InsertAfter",
    },
  },
```

**Reply by gpxl-dev** [reply]
cc @seb-hyland

**Reply by kenandotfyi** [reply]
> I added this keybind to get the same with just i and a 
> 
> ```json
> 
>   {
> 
>     "context": "Editor && vim_mode == visual",
> 
>     "bindings": {
> 
>       "i": "vim::InsertBefore",
> 
>       "a": "vim::InsertAfter",
> 
>     },
> 
>   },
> 
> ```
> 
> 

Oh yeah that's good too👍

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
